anvil/src/shell.rs

```diff
@@ -100,6 +151,45 @@ pub fn init_shell(
                 position: (10, 10),
                 serial: 42,
             }),
+            XdgRequest::Move {
+                surface,
+                seat,
+                serial,
+            } => {
+                let seat = Seat::from_resource(&seat).unwrap();
+                // TODO: touch move.
+                let pointer = seat.get_pointer().unwrap();
+
+                // Check that this surface has a click grab.
+                if !pointer.has_grab(serial) {
+                    return;
+                }
+
+                let start_data = pointer.grab_start_data().unwrap();
+
+                // If the focus was for a different surface, ignore the request.
+                if start_data.focus.is_none()
+                    || !compositor_token.contains_in_tree(
+                        surface.get_surface().unwrap(),
+                        &start_data.focus.as_ref().unwrap().0,
+                    )
+                {
+                    return;
+                }
```

Comment:
Why do we need this check? I'd say checking that the serial matches the grab & that the focused surface belongs to the correct client should be enough.

Reply:
Well, this is the "focused surface belongs to the correct client" part of the check. It verifies that the grab's focused surface is the surface that send Move, or one of its children.

Reply:
I think a looser check like would be sufficient:

```rust
let same_client = start_data.focus().as_ref().map(
    |sdata| sdata.as_ref().same_client_as(surface.get_surface())
).unwrap_or(false);
if !same_client {
    return;
}
```

I don't see why we should prevent a client from moving ones of its windows using a grab initiated on an other window, as long as both windows belong to the same client.